README.md

```diff
@@ -2,31 +2,230 @@
+| LICENSE        | Text file with the license how to use the source code of DSP-TOOLS              |
+| Makefile       | Definition of commands that can be executed with `make [command]`               |
+| poetry.lock    | Pinned versions of all (sub-) dependencies, allows a deterministic installation |
+| mkdocs.yml     | Configuration of `mkdocs`, used to build the documentation webpage              |
+
+In earlier times, there were some more configuration files, but thanks to poetry, they are not necessary anymore:
+
+| Deprecated file      | Replaced by                             | Replaced by                                          |
+|----------------------|-----------------------------------------|------------------------------------------------------|
+| MANIFEST.in          | files to include into distribution      | pyproject.toml: [tool.poetry.include]                |
+| setup.py             | project metadata, dependencies          | pyproject.toml                                       |
+| setup.cfg            | configuration for setuptools            | pyproject.toml                                       |
+| requirements.txt     | all (sub-) dependencies                 | pyproject.toml: [tool.poetry.dependencies]           |
+| dev-requirements.txt | additional dependencies for development | pyproject.toml: [tool.poetry.group.dev.dependencies] |
+| Pipfile              | direct dependencies                     | pyproject.toml: [tool.poetry.dependencies]           |
+| Pipfile.lock         | pinned dependencies                     | poetry.lock                                          |
+
+
+
+### Dependency management
+
+#### Historical considerations
+
+The classic way to manage the dependencies was to write the required packages by hand into a `requirements.txt` and 
+into a `setup.py` file. 
+
+But this is cumbersome and error-prone, so there was a time when [pipenv](https://pipenv.pypa.io/en/latest/) was the 
+way to go: Pipenv introduced the important distinction between (a) dependencies necessary to run the application, 
+(b) dependencies necessary for development, and (c) sub-dependencies, i.e. dependencies of your dependencies. Another 
+useful concept of pipenv is the distinction between a human-friendly list of (mostly unpinned) direct dependencies and 
+a machine-friendly definition of exact (pinned) versions of all dependencies.  
+But since pipenv has no packaging functionality, it was necessary to sync the dependency definitions from `Pipfile` to  
+`requirements.txt` and `setup.py`.  
+
+`setup.py`, too, is problematic, especially 
+[calling `setup.py sdist bdist_wheel`](https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html#summary). 
+Python projects should define their dependencies and metadata in the modern `pyproject.toml` file. So it is 
+necessary to dynamically manage the dependencies in `pyproject.toml`. And poetry seems to be the only tool capable 
+of doing this.
```

Comment:
Same as above, who would be the target user of this kind of documentation? I see that this can be interesting to know, but IMHO irrelevant to someone who wants to work with DSP-TOOLS.